docs/slurk_bots_events.rst

```diff
@@ -0,0 +1,228 @@
+- ``room(int)``: the ``id`` of the room that was entered by this user
+
+Task bots are generally sent to rooms to instruct users and provide resources necessary to the task fulfillment. The ``joined_room`` event handler could be used to introduce the bot to the users and set an initial task description.
+
+.. code-block:: python
+
+    @self.sio.event
+    def left_room(data):
+        do_something(data)
+
+``data`` in ``left_room`` has this structure:
+
+- ``user(int)``: the ``id`` of the user who caused this event
+- ``room(int)``: the ``id`` of the room that was left by this user
+
+Monitoring overall movement
+---------------------------
+Bots are also notified once a user joins or leaves one of the rooms the bot is placed in. The term `user` here includes the bot itself as well as other bots and human users.
+
+
+.. code-block:: python
+
+    @self.sio.event
+    def status(data):
+        do_something(data)
+
+``data`` in ``status`` has this structure:
+
+- ``type(str)``: the status type, either `join` or `leave`
+- ``user(dict)``: dictionary of ``id(int)`` and ``name(str)`` of the user who caused this event
+- ``room(int)``: the ``id`` of the room that was entered or left, respectively
+- ``timestamp(str)``: as ISO 8601: ``YYYY-MM-DD hh:mm:ss.ssssss`` in UTC Time
+
+Chat
+~~~~
+All of the events mentioned below can be either ``private`` or not. If an event is ``private`` it is only sent to a designated receiver. If this receiver is the bot, it receives the event. Otherwise it does not receive it. If an event is not ``private`` it can be seen by all users in the specified room.
+Only bots are intended to send private content but for debugging purposes you can use the chat interface and the following syntax to send private messages ``@<user_id> <text>`` or private images ``@<user_id> image: <url>``. Make sure that whoever is supposed to send private content is assigned the ``send_privately`` permission.
+
+Messages
+--------
+Every data collection experiment evolves around users exchanging messages. Those can be sent by any user that is assigned the ``send_message`` or ``send_html_message`` permission. A bot may wish to verify message content, count messages until a certain milestone is reached or otherwise process user messages.
+Messages can also be sent by bots:
+
+.. code-block:: python
+
+    self.sio.emit(
+        "text",
+        data
+    )
+
+``data`` is a dictionary with the following keys:
+
+- ``message(str)``: the content of the text message
+- ``room(int)``: the ``id`` of the room where the text message will be sent to
+- ``receiver_id(int, optional)``: the ``id`` of the user that this message is directed at
+- ``broadcast(bool, optional)``: ``True`` if the message should be transmitted to all connected users. ``False`` otherwise
+- ``html(bool, optional)``: ``True`` if special html formatting should be applied to a message. This requires ``send_html_message`` permissions. ``False`` otherwise.

```

Comment:
```suggestion
- ``html(bool, optional)``: ``True`` if special html formatting should apply to a message. This requires ``send_html_message`` permissions. ``False`` otherwise.
```